### PR TITLE
Signal Handler for Souffle / Fix issue #236 and issue #360

### PR DESCRIPTION
**Pull request by b-scholz**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -51,6 +52,8 @@
 #include <sys/stat.h>
 #include <unistd.h>
 
+SignalHandler* SignalHandler::m_singleton = nullptr;
```
Comment (mmcgr) [inline src/main.cpp]: Can we use a static variable in SignalHandler::instance() instead? Then the SignalHandler code can handle its own setup.

File: src/main.cpp [hunk 2]
```diff
@@ -30,6 +30,7 @@
 #include "RamExecutor.h"
 #include "RamStatement.h"
 #include "RamTranslator.h"
+#include "SignalHandler.h"
```
Comment (mmcgr) [inline src/main.cpp]: Is this include still needed?

File: src/SignalHandler.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ * Class SignalHandler captures signals
+ * and reports the context where the signal occurs.
+ * The signal handler is implemented as a singleton.
+ */
+class SignalHandler {
+private:
+    // signal context information
+    std::atomic<const char*> msg;
+
+private:
```
Comment (mmcgr) [inline src/SignalHandler.h]: This section is already private.

**Comment by b-scholz** [comment]
Martin - when you do the merge pull request can you click on the option rebase. I believe that would be better than having my local multiple commits.

**Comment by b-scholz** [comment]
I noticed  you can only do squash and merge. Can you try this?

**Pull request merged**